src/nix.rs

```diff
@@ -194,22 +198,27 @@ impl CallOpts {
     /// # env::set_var("NIX_PATH", "nixpkgs=./nix/bogus-nixpkgs/");
     ///
     /// let tempdir = tempfile::tempdir().unwrap();
-    /// let location = nix::CallOpts::expression(r#"
+    /// let (location, gc_root) = nix::CallOpts::expression(r#"
     ///             import <nixpkgs> {}
     /// "#)
     ///         .attribute("hello")
-    ///         .path(&tempdir.path())
+    ///         .path()
     ///         .unwrap()
-    ///         .into_os_string()
-    ///         .into_string().unwrap()
     ///         ;
     ///
+    /// let location = location.into_os_string().into_string().unwrap();
     /// println!("{:?}", location);
     /// assert!(location.contains("/nix/store"));
     /// assert!(location.contains("hello-"));
+    /// drop(gc_root);
     /// ```
     ///
-    /// Note, path() return an error if there are multiple paths
+    /// `path` returns a handle to a temporary GC root directory
+    /// (`gc_root` in the example above). Until that is dropped,
```

Comment:
It doesn't return the directory, it returns an opaque value which keeps the paths alive.